district_data_import.py

```diff
@@ -144,6 +146,53 @@
             'ZIP': district_record['zip']
         })
 
+        # Update for potential switch of parent name order
+
+        curr_informal_sal = dp_donorrecord['INFORMAL_SAL']
+        curr_main_f_name = dp_donorrecord['FIRST_NAME']
+        curr_spouse_f_name = dp_donorrecord['SP_FNAME']
+
+        dp_donorrecord_with_pnames = district_data_utils.create_dp_donorrecord(district_record=district_record, school_year=args.school_year)
+        dp_donorrecord.update({
+            'FIRST_NAME': dp_donorrecord_with_pnames['FIRST_NAME'],
+            'LAST_NAME': dp_donorrecord_with_pnames['LAST_NAME'],
+            'SP_FNAME': dp_donorrecord_with_pnames['SP_FNAME'],
+            'SP_LNAME': dp_donorrecord_with_pnames['SP_LNAME'],
+            'EMAIL': dp_donorrecord_with_pnames['EMAIL'],
+            'SPOUSE_EMAIL': dp_donorrecord_with_pnames['SPOUSE_EMAIL'],
+            'MOBILE_PHONE': dp_donorrecord_with_pnames['MOBILE_PHONE'],
+            'SPOUSE_MOBILE': dp_donorrecord_with_pnames['SPOUSE_MOBILE'],
+            'SALUTATION': dp_donorrecord_with_pnames['SALUTATION'],
+            'OPT_LINE': dp_donorrecord_with_pnames['OPT_LINE']
+        })
+
+        # For informal salutations, we update it only if it is a straightforward switch of the parent name order. 
+        # If the computed value is not a straightforward switch, it indicates that a manual update may have occured based on 
+        # personal knowledge of nicknames and such, so we leave it alone. 
+
+        def create_informal_sal(main_f_name, spouse_f_name):
+            if len(spouse_f_name) != 0:
+                informal_sal = main_f_name + " and " + spouse_f_name
+            else:
+                informal_sal = main_f_name
+            return informal_sal
+
+        row = utils.dict_filtered_copy(dp_donorrecord, informal_sal_headers)
+        row['_CURR_INFORMAL_SAL'] = curr_informal_sal
+        # Cannot use the conditional below as import may have already occured. 
```

Comment:
This comment sounds like you have a mix of one-time and permanent logic in the script. Let's keep only permanent logic on the master branch.